tests/files/locations/test_s3.py

```diff
@@ -0,0 +1,40 @@
+import pytest
+
+from astro.files.locations import location_factory
+
+
+def describe_is_valid_path():
+    """test is_valid_path() method"""
+
+    def with_supported_filepaths():  # skipcq: PTC-W0065, PYL-W0612
+        """With supported file paths"""
+        filepath = "s3://bucket/some-file"
+        location = location_factory(filepath)
+        assert location.is_valid_path(filepath) is True
+
+
+def describe_get_transport_params():
+    """test get_transport_params() method"""
+
+    def with_s3():  # skipcq: PYL-W0612, PTC-W0065
+        """with S3 filepath"""
+        path = "s3://bucket/some-file"
+        location = location_factory(path)
+        credentials = location.get_transport_params()
+        assert "botocore.client.S3" in str(credentials["client"].__class__)
+
+
+@pytest.mark.integration
```

Comment:
This can be converted to unit test and mock results, we don't need an integration test for it.

Reply:
Done.